packages/vue/src/components/range/MultiRange.jsx

```diff
@@ -0,0 +1,252 @@
+import { Actions, helper } from '@appbaseio/reactivecore';
+import VueTypes from 'vue-types';
+import Title from '../../styles/Title';
+import Container from '../../styles/Container';
+import { UL, Checkbox } from '../../styles/FormControlList';
+import { connect } from '../../utils/index';
+import types from '../../utils/vueTypes';
+
+const {
+  addComponent,
+  removeComponent,
+  watchComponent,
+  updateQuery,
+  setQueryListener
+} = Actions;
+
+const { isEqual, checkValueChange, getClassName } = helper;
+const MultiRange = {
+  name: 'MultiRange',
+  data() {
+
+    this.state = {
+      currentValue: [],
+      showModal: false,
+      // selectedValues hold the selected items as keys for O(1) complexity
+      selectedValues: {},
+    };
+    this.type = 'range';
+    this.locked = false;
+    return this.state;
+  },
+  props: {
+
+    beforeValueChange: types.func,
+    className: VueTypes.string.def(''),
+    componentId: types.stringRequired,
+    customQuery: types.func,
+    data: types.data,
+    dataField: types.stringRequired,
+    defaultSelected: types.string,
+    filterLabel: types.string,
+    innerClass: types.style,
+    react: types.react,
+    showFilter: VueTypes.bool.def(true),
+    showCheckbox: VueTypes.bool.def(true),
+    title: types.title,
+    URLParams: VueTypes.bool.def(false)
+  },
+  methods: {
+    setReact(props) {
+      if (props.react) {
+        props.watchComponent(props.componentId, props.react);
+      }
+    },
+    handleChange(e) {
+      this.selectItem(e.target.value);
+    },
+    selectItem(item, isDefaultValue = false, props = this.$props) {
+      // ignore state updates when component is locked
+      if (props.beforeValueChange && this.locked) {
+        return;
+      }
+
+      this.locked = true;
+      let { currentValue, selectedValues } = this.state;
+
+      if (!item) {
+        currentValue = [];
+        selectedValues = {};
+      } else if (isDefaultValue) {
+
+        // checking if the items in defaultSeleted exist in the data prop
+        currentValue = MultiRange.parseValue(item, props);
+        currentValue.forEach((value) => {
+          selectedValues = { ...selectedValues, [value.label]: true };
+        });
+      } else if (selectedValues[item]) {
+
+        currentValue = currentValue.filter(value => value.label !== item);
+        const { [item]: del, ...selected } = selectedValues;
+        selectedValues = selected;
+      } else {
+
+        const currentItem = props.data.find(value => item === value.label);
+        currentValue = [...currentValue, currentItem];
+        selectedValues = { ...selectedValues, [item]: true };
+      }
+      const performUpdate = () => {
+        this.state.currentValue = currentValue;
+        this.state.selectedValues = selectedValues;
+        this.updateQueryHandler(currentValue, props);
+        this.locked = false;
+        this.$emit('valueChange', 'divyanshu');
```

Comment:
What are you (`divyanshu`) doing here?
Please fix all these details, I'll start reviewing it back.

Reply:
@bietkul sorry it was just for testing, I mistakely added it here, will update the PR.